lib/streamlit/runtime/pages_manager.py

```diff
@@ -237,6 +232,13 @@ def __init__(
         self._intended_page_script_hash: PageHash | None = None
         self._intended_page_name: PageName | None = None
         self._current_page_script_hash: PageHash = ""
+        # A relic of v1 of Multipage apps, we performed special handling
+        # for apps with a pages directory. We will keep this flag around
+        # for now to maintain the behavior for apps that were created with
+        # the pages directory feature.
+        PagesManager.uses_pages_directory = Path(
```

Comment:
nit: maybe add as a comment the flag might be deactivated again if there is another `st.navigation` call indicating usage of MPAv2